classes/submission/reviewAssignment/Collector.php

```diff
@@ -0,0 +1,298 @@
+ */
+
+namespace PKP\submission\reviewAssignment;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+use PKP\core\Core;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\submission\ViewsCount;
+
+/**
+ * @template T of ReviewAssignment
+ */
+class Collector implements CollectorInterface, ViewsCount
+{
+    public DAO $dao;
+    public ?array $contextIds = null;
+    public ?array $submissionIds = null;
+    public bool $isIncomplete = false;
+    public bool $isArchived = false;
+    public bool $isOverdue = false;
+    public ?array $reviewRoundIds = null;
+    public ?array $reviewerIds = null;
+    public bool $isLastReviewRound = false;
+    public ?int $count = null;
+    public ?int $offset = null;
+    public ?int $reviewMethods = null;
+    public ?int $stageId = null;
+    public ?array $reviewFormIds = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /** @copydoc DAO::getCount() */
+    public function getCount(): int
+    {
+        return $this->dao->getCount($this);
+    }
+
+    /**
+     * @copydoc DAO::getIds()
+     * @return Collection<int,int>
+     */
+    public function getIds(): Collection
+    {
+        return $this->dao->getIds($this);
+    }
+
+    /**
+     * @copydoc DAO::getMany()
+     * @return LazyCollection<int,T>
+     */
+    public function getMany(): LazyCollection
+    {
+        return $this->dao->getMany($this);
+    }
+
+    /**
+     * Filter review assignments by one or more contexts
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Filter review assignments by associated submissions
+     */
+    public function filterBySubmissionIds(?array $submissionIds): self
+    {
+        $this->submissionIds = $submissionIds;
+        return $this;
+    }
+
+    /**
+     * Filter by active assignments only:
+     * is not completed or declines or cancelled and associated submission is in the review stage
+     *
+     */
+    public function filterByIsIncomplete(?bool $isIncomplete): self
+    {
+        $this->isIncomplete = $isIncomplete;
+        return $this;
+    }
+
+    /**
+     * Filter by completed or declined assignments
+     */
+    public function filterByIsArchived(?bool $isArchived): self
+    {
+        $this->isArchived = $isArchived;
+        return $this;
+    }
+
+    /**
+     * Filter by overdue assignments
+     */
+    public function filterByIsOverdue(?bool $isOverdue): self
+    {
+        $this->isOverdue = $isOverdue;
+        return $this;
+    }
+
+    /**
+     * Filter by review round ids
+     */
+    public function filterByReviewRoundIds(?array $reviewRoundIds): self
```

Comment:
Better to type-hint a `static` return type, rather than `self`, I think: https://wiki.php.net/rfc/static_return_type